packages/app/src/runner/SpecRunnerContainerOpenMode.vue

```diff
@@ -30,13 +29,21 @@ const specs = computed(() => {
 
 watchSpec(specs)
 
-watchEffect(() => {
-  const specPath = router.currentRoute.value.query.file
+specStore.$subscribe((mutation, state) => {
+  const file = route.query.file as string
 
-  if (specPath && !specStore.activeSpec) {
-    router.push({ name: 'Specs', params: {
-      unrunnable: router.currentRoute.value.query.file as string,
-    } })
+  const shouldRedirect = route.name === 'SpecRunner' && file && state.activeSpec === null
+
```

Comment:
Checking for `null` here because I added `undefined` as the initial state in the store. That might be unnecessary now with the subscribe, but feels OK to leave this in so that the state of the initial value is different from the state when some event explicitly sets the `activeSpec` to null.

Reply:
I think the whole notion of undefined/null having different meanings is generally not a good idea, since your entire team needs to remember the difference. I'm okay with this for now, but we should likely standardize. An example of this would be the TypeScript (and VS Code iirc) code bases - they exclusively use `undefined`, and everything is simple for that reason.

Any reason we can't just used `undefined` here (all the rest of `packages/app`?